hack/cluster.sh

```diff
@@ -90,14 +96,24 @@ main() {
 	local op="$1"
 	shift
 	cd "$PROJECT_ROOT"
-
-	# NOTE: all operations are relative to tmp
+	export PATH="$BIN_DIR:$PATH"
 	mkdir -p "${TMP_DIR}"
-	git_checkout
+	ensure_all_tools
```

Comment:
Do we need to call ensure_all_tools at every operation? I think we should call this during cluster_prereqs step or during on_cluster_up. Wdyt?

Reply:
yes, always best to ensure all tools are present. This is an assumption that all functions can make. thus no need to check if `x, y, z` is installed.

Reply:
Why need this in case of cluster_down?

Reply:
The same applies,  ... it is best to ensure all tools exist before starting anything and for a non - production script, these things do not matter.